inc/config/query-runner/query-runner.php

```diff
@@ -99,22 +99,23 @@ public function execute( array $input_variables ): array|WP_Error {
 		}
 
 		// The body is a stream... if we need to read it in chunks, etc. we can do so here.
-		$response_data = $response->getBody()->getContents();
+		$raw_response_data = $response->getBody()->getContents();
 
-		$is_collection = $this->query_context->output_variables['is_collection'] ?? false;
-
-		if ( isset( $response_data['errors'][0]['message'] ) ) {
+		if ( isset( $raw_response_data['errors'][0]['message'] ) ) {
 			$logger = LoggerManager::instance();
-			$logger->warning( sprintf( 'Query error: %s', esc_html( $response_data['errors'][0]['message'] ) ) );
+			$logger->warning( sprintf( 'Query error: %s', esc_html( $raw_response_data['errors'][0]['message'] ) ) );
 		}
 
+		// Optionally process the raw response data using query context custom logic.
+		$response_data = $this->query_context->process_response( $raw_response_data, $input_variables );
+
 		// This method always returns an array, even if it's a single item. This
 		// ensures a consistent response shape. The requestor is expected to inspect
 		// is_collection and unwrap if necessary.
-		$results = $this->map_fields( $response_data, $is_collection );
+		$results = $this->map_fields( $response_data );
```

Comment:
Removing arg `$is_collection` means that `is_collection()` gets called twice. Can we get it back to one call?